x-pack/plugins/rollup/public/crud_app/sections/job_list/detail_panel/detail_panel.test.js

```diff
@@ -0,0 +1,334 @@
+    let findTestSubject;
+
+    beforeEach(() => {
+      ({ component, findTestSubject } = initTestBed());
+    });
+
+    it('should have the title set to the current Job "id"', () => {
+      const { job } = defaultProps;
+      const title = component.find('#rollupJobDetailsFlyoutTitle').hostNodes();
+      expect(title.length).toBeTruthy();
+      expect(title.text()).toEqual(job.id);
+    });
+
+    it('should have children if it\'s open', () => {
+      expect(component.find('DetailPanelUi').children().length).toBeTruthy();
+    });
+
+    it('should *not* have children if it\s closed', () => {
+      ({ component } = initTestBed({ isOpen: false }));
+      expect(component.find('DetailPanelUi').children().length).toBeFalsy();
+    });
+
+    it('should show a loading when the job is loading', () => {
+      ({ component, findTestSubject } = initTestBed({ isLoading: true }));
+      const loading = findTestSubject('rollupJobDetailLoading');
+      expect(loading.length).toBeTruthy();
+      expect(loading.text()).toEqual('Loading rollup job...');
+
+      // Make sure the title and the tabs are visible
+      expect(component.find('EuiTab').length).toBeTruthy();
+      expect(component.find('#rollupJobDetailsFlyoutTitle').length).toBeTruthy();
+    });
+
+    it('should display a message when no job is provided', () => {
+      ({ component, findTestSubject } = initTestBed({ job: undefined }));
+      expect(findTestSubject('rollupJobDetailFlyout').find('EuiFlyoutBody').text()).toEqual('Rollup job not found');
+    });
+  });
+
+  describe('tabs', () => {
+    const tabActive = JOB_DETAILS_TAB_SUMMARY;
+    const { component } = initTestBed({ panelType: tabActive });
+    const tabs = component.find('EuiTab');
+    const getTab = (id) => {
+      let selectedTab;
+      tabs.forEach((tab) => {
+        if (tab.text() === tabToHumanizedMap[id].props.defaultMessage) {
+          selectedTab = tab;
+        }
+      });
+      return selectedTab;
+    };
+
+    it('should have 5 tabs visible', () => {
+      const expected = JOB_DETAILS_TABS.map(id => tabToHumanizedMap[id].props.defaultMessage);
+      const tabsLabel = tabs.map(tab => tab.text());
+
+      expect(tabsLabel).toEqual(expected);
+    });
+
+    it('should set default selected tab to the "panelType" prop provided', () => {
+      const tab = getTab(tabActive);
+      expect(tab.props().isSelected).toEqual(true);
+    });
+
+    it('should select the tab when clicking on it', () => {
+      const { job, openDetailPanel } = defaultProps;
+      const termsTab = getTab(JOB_DETAILS_TAB_TERMS);
+
+      expect(termsTab.props().isSelected).toEqual(false);
+
+      termsTab.simulate('click');
+
+      expect(openDetailPanel.mock.calls.length).toBe(1);
+      expect(openDetailPanel.mock.calls[0][0]).toEqual({
+        jobId: job.id,
+        panelType: JOB_DETAILS_TAB_TERMS
+      });
+    });
+  });
+
+  describe('job detail', () => {
+    describe('summary tab content', () => {
```

Comment:
I think it's good that we've explored this direction. But It seems like maintaining this level of fine-grained testing for this particular suite will become a burden as we make changes to the UI. I think we'd be better off using snapshots for each of these tabs. 

```js
      it('renders logistics, date histogram, and stats sections', () => {
        expect(findTestSubject('rollupJobDetailsSummaryTabContent')).toMatchSnapshot();
      });
```

The downside is that this single test results in an 800-line snapshot, which is difficult to read and may change as EUI's implementation details change. But the snapshot doesn't need to be readable, we just need to be able to verify that the UI fulfills our goals in the browser, and then freeze it that way as a snapshot. If the tests fail due to a code change, we just need to make sure it's good in the browser and update the snapshot. And If the tests fail due to an EUI change, we'd just check the browser and update the snapshot.

Another benefit of a snapshot is that it will also catch changes when we _add_ content. These current tests already know what they're looking for, so they'll miss any new sections or subsections which get added.

Want to discuss this over Zoom?

Reply:
Yes, I think we better discuss this over Zoom. 😊 but here are my thoughts.  
To me, there is no _intent_ behind a snapshot. When I read this test

```js
expect(findTestSubject('rollupJobDetailsSummaryTabContent')).toMatchSnapshot();
```

I don't have any idea of what is being tested, what is the expected behavior of the component, what it should display, what has been agreed on when it was written.

About maintainance, it should not be a burden IMO, to the contrary. The product team comes and says, "_hey could we add another section to the summary tab with that and that value_". And in good TDD (😊 ) we'd come to the test file, add the expected section and the expected values and then update the UI.